edi_stock_oca/models/stock_picking.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 OCA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _name = "stock.picking"
+
+    _inherit = ["stock.picking", "edi.exchange.consumer.mixin"]
+
+    edi_auto_disabled = fields.Boolean(
+        states={
+            "draft": [("readonly", False)],
+            "waiting": [("readonly", False)],
+            "confirmed": [("readonly", False)],
+            "assigned": [("readonly", False)],
+        },
+    )
+
+    def button_validate(self):
+        res = super().button_validate()
+        self._event("on_validate").notify(self)
+        return res
```

Comment:
Wouldn't it be better to listen at `action_done`?, in button_validate there are several events that can pop before actually doing the picking (backorder wizard, no quantity done wizard, etc.).

Reply:
@LoisRForgeFlow Can you recheck it please :smile: ?